crates/ty_ide/src/docstring.rs

```diff
@@ -0,0 +1,628 @@
+                        && section_name
+                            .chars()
+                            .all(|c| c.is_alphabetic() || c.is_whitespace())
+                    {
+                        // Check if this is a known section name
+                        let known_sections = [
+                            "Returns", "Return", "Raises", "Yields", "Yield", "Examples",
+                            "Example", "Note", "Notes", "Warning", "Warnings",
+                        ];
+                        if known_sections.contains(&section_name) {
+                            if let Some(param_name) = current_param.take() {
+                                param_docs.insert(param_name, current_doc.trim().to_string());
+                                current_doc.clear();
+                            }
+                            in_args_section = false;
+                            continue;
+                        }
+                    }
+                }
+            }
+
+            if let Some(captures) = param_regex.captures(line) {
+                // Save previous parameter if exists
+                if let Some(param_name) = current_param.take() {
+                    param_docs.insert(param_name, current_doc.trim().to_string());
+                    current_doc.clear();
+                }
+
+                // Start new parameter
+                if let (Some(param), Some(desc)) = (captures.get(1), captures.get(3)) {
+                    current_param = Some(param.as_str().to_string());
+                    current_doc = desc.as_str().to_string();
+                }
+            } else if line.starts_with(' ') || line.starts_with('\t') {
+                // This is a continuation of the current parameter documentation
+                if current_param.is_some() {
+                    if !current_doc.is_empty() {
+                        current_doc.push('\n');
+                    }
+                    current_doc.push_str(line.trim());
+                }
+            } else {
+                // This is a line that doesn't start with whitespace and isn't a parameter
+                // It might be a section or other content, so stop processing args
+                if let Some(param_name) = current_param.take() {
+                    param_docs.insert(param_name, current_doc.trim().to_string());
+                    current_doc.clear();
+                }
+                in_args_section = false;
+            }
+        }
+    }
+
+    // Don't forget the last parameter
+    if let Some(param_name) = current_param {
+        param_docs.insert(param_name, current_doc.trim().to_string());
+    }
+
+    if param_docs.is_empty() {
+        None
+    } else {
+        Some(param_docs)
+    }
+}
+
+/// Calculate the indentation level of a line (number of leading whitespace characters)
+fn get_indentation_level(line: &str) -> usize {
+    line.chars().take_while(|c| c.is_whitespace()).count()
+}
```

Comment:
Nit: You could use https://github.com/astral-sh/ruff/blob/9f3a38d408f473df7a6b3574c5d1389bef303dd5/crates/ruff_python_trivia/src/whitespace.rs#L49-L52

unless you want to account for unicode whitespace characters too